src/main/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdk.java

```diff
@@ -176,15 +181,49 @@ public SdkUpdater newUpdater() {
   }
 
   /** Get a new {@link ManagedCloudSdk} instance for @{link Version} specified. */
-  public static ManagedCloudSdk newManagedSdk(Version version) throws UnsupportedOsException {
+  public static ManagedCloudSdk newManagedSdk(Version version)
+      throws UnsupportedOsException, InvalidOsStateException {
+    OsInfo osInfo = OsInfo.getSystemOsInfo();
     return new ManagedCloudSdk(
         version,
-        Paths.get(System.getProperty("user.home"), ".google-cloud-tools-java", "managed-cloud-sdk"),
-        OsInfo.getSystemOsInfo());
+        getOsSpecificManagedSdkHome(osInfo.name(), System.getProperties(), System.getenv()),
+        osInfo);
   }
 
   /** Convenience method to obtain a new LATEST {@link ManagedCloudSdk} instance. */
-  public static ManagedCloudSdk newManagedSdk() throws UnsupportedOsException {
+  public static ManagedCloudSdk newManagedSdk()
+      throws UnsupportedOsException, InvalidOsStateException {
     return newManagedSdk(Version.LATEST);
   }
+
+  @VisibleForTesting
+  static Path getOsSpecificManagedSdkHome(
```

Comment:
This feels like a no-args methods. The calls to System.getenv etc. can be made here.

Reply:
I think the arguments are being passed to allow simulate testing for different OSs